arch/toolchains/ecmwf-hpc2020-nvhpc.cmake

```diff
@@ -16,15 +16,19 @@ set( ECBUILD_FIND_MPI ON )
 # OpenMP FLAGS
 ####################################################################
 
-set( OpenMP_C_FLAGS             "-mp -mp=bind,allcores,numa" )
-set( OpenMP_CXX_FLAGS           "-mp -mp=bind,allcores,numa" )
-set( OpenMP_Fortran_FLAGS       "-mp -mp=bind,allcores,numa" )
+# Note: OpenMP_<lang>_FLAGS gets overwritten by the FindOpenMP module
+# unless its stored as a cache variable
+set( OpenMP_C_FLAGS             "-mp -mp=bind,allcores,numa" CACHE STRING "" FORCE )
+set( OpenMP_CXX_FLAGS           "-mp -mp=bind,allcores,numa" CACHE STRING "" FORCE )
+set( OpenMP_Fortran_FLAGS       "-mp -mp=gpu,bind,allcores,numa" CACHE STRING "" FORCE )
 
 ####################################################################
 # OpenAcc FLAGS
 ####################################################################
 
-set( OpenACC_Fortran_FLAGS "-acc=gpu -gpu=lineinfo,fastmath" )
+# NB: We have to add `-mp` again to avoid undefined symbols during linking
+# (smells like an Nvidia bug)
+set( OpenACC_Fortran_FLAGS "-acc=gpu -mp=gpu -gpu=cc80,lineinfo,fastmath" )
```

Comment:
To be consistent with OpenMP flags we could also add it to the cache and allow to override via command-line.

Reply:
Agreed, hadn't done it because it wasn't necessary but makes sense absolutely.